### PR TITLE
Improve dto mapper.

### PR DESCRIPTION
**Pull request by mshima**
- look at everything back relationship even if the entity does not contains the forward relationship.
Needed for completeness.
- implements another priority called prepararingRelationship.
at `default` priority, otherRelationships needs to be settled for a complete back reference.
- adds collection mapping.

Related to https://github.com/jhipster/generator-jhipster/issues/12844
Fixes https://github.com/jhipster/generator-jhipster/issues/13036.

---

Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Comment by pascalgrimaud** [comment]
will it impact composite keys feature ?
as composite keys is already a big pull request, I'd prefer to merge it first, to avoid more conflict there
What do you think @mshima ?

**Comment by mshima** [comment]
@pascalgrimaud IMO trying to land composite id at once is REALLY difficult.

I've created this PR and the entity back reference PR because I was having problem making https://github.com/jhipster/generator-jhipster/pull/12971 pass CI, in the end I had to adjust all 3 PRs.
I think this PR is a step to composite id, it aims for integration of both sides of a relationship.

Current implementation of composite id changes from a parallel composing to an eager function that prepare every entity at once, changing our workflow completely. I am not in favor of merging it this way, blueprints would have a hard time to customise it.
I would say there is a long run yet.

File: .github/workflows/webflux.yml [hunk 1]
```diff
@@ -126,7 +126,7 @@ jobs:
                     #                      war: 0
                     #                      e2e: 1
                     - app-type: webflux-psql
-                      entity: sql
+                      entity: sqlfull
```
Comment (mshima) [inline .github/workflows/webflux.yml:129]:
```suggestion
                      entity: sql
```

**Comment by pascalgrimaud** [comment]
@mshima : is it ready to be merged ? As I don't have time to review in depth, I'm trusting you here

**Comment by mshima** [comment]
@pascalgrimaud it is.

**Comment by mshima** [comment]
@pascalgrimaud thanks.

**Pull request merged**